src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -136033,6 +136034,236 @@ partial class C2<T> : Base<
                 );
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/40538")]
+        public void AnnotationMismatch_Interface_GenericClass_01()
+        {
+            var source = """
+                #nullable enable
+                var c = new C<int>();
+
+                partial class C<T> : I<T> { }
+
+                partial class C<T> :
+                    #nullable disable
+                    I<T>
+                    #nullable enable
+                {
+                }
+
+                public interface I<T> { }
+                """;
+
+            var verifier = CompileAndVerify(CreateCompilationWithMscorlib40(source)).VerifyDiagnostics();
+
+            verifier.VerifyIL("<top-level-statements-entry-point>", """
+                {
+                  // Code size        7 (0x7)
+                  .maxstack  1
+                  IL_0000:  newobj     "C<int>..ctor()"
+                  IL_0005:  pop
+                  IL_0006:  ret
+                }
+                """);
+
+            verifier.VerifyTypeIL("C`1", """
+                .class private auto ansi beforefieldinit C`1<T>
+                	extends [mscorlib]System.Object
+                	implements class I`1<!T>
+                {
+                	.param type T
+                		.custom instance void System.Runtime.CompilerServices.NullableAttribute::.ctor(uint8) = (
+                			01 00 02 00 00
```

Comment:
I don't know how the NullableAttributeEncoding works well enough to interpret this. Is this saying that the T is nullable-enabled?

It may be more clear to use a 'symbolValidator' which uses the symbol APIs to show the nullability of the T.